internal/common/thrift_util.go

```diff
@@ -48,26 +42,11 @@ func TListSerialize(ts []thrift.TStruct) (b []byte, err error) {
 	ctx := context.Background()
 	for _, v := range ts {
 		if e := v.Write(ctx, t.Protocol); e != nil {
-			err = thrift.PrependError("error writing TStruct: ", e)
-			return
+			return nil, thrift.PrependError("error writing TStruct: ", e)
 		}
 	}
 
-	if err = t.Protocol.Flush(ctx); err != nil {
-		return
-	}
-
-	if err = t.Transport.Flush(ctx); err != nil {
-		return
-	}
-
-	b = t.Transport.Bytes()
-	return
-}
-
-// TDeserialize is used to deserialize []byte to thrift TStruct
-func TDeserialize(ctx context.Context, t thrift.TStruct, b []byte) (err error) {
-	return thrift.NewTDeserializer().Read(ctx, t, b)
+	return t.Transport.Bytes(), t.Protocol.Flush(ctx)
```

Comment:
this (and not resetting) is a rather significant set of changes - is there something that implies it's correct?  e.g. newer thrift/tchannel docs?

Reply:
I've looked into thrift.NewTSerializer() and it uses InMemoryTransport that has a comment of Flush is Noop for it, since it uses plain buffer.
Also, Protocol.Flush is just a wrapper around Transport.Flush().

Reply:
```
// Flushing a memory buffer is a no-op
func (p *TMemoryBuffer) Flush(ctx context.Context) error {
	return nil
}
```

Reply:
We always initialize one buffer with
```
t := thrift.NewTSerializer()
```
Why it should be reset?

Reply:
Why would it be reset: I agree it doesn't _look_ correct, but the only thing that matters is if it _is_ correct.  If that hasn't been verified, it needs to be, but hasn't been mentioned anywhere so I'm not sure if it has been.